### DISCUSSION TITLE
Data validation list type formulae limit - drop-down cell items limit

### DISCUSSION THREAD

**Discussion by DevMoustafa97** [exceljs/exceljs#2069] [Q&A]
If we specify a list of valid values (One, Two, Three, Four).
Excel will provide a dropdown with these values using the code below

```
worksheet.getCell('A1').dataValidation = {
  type: 'list',
  allowBlank: true,
  formulae: ['"One,Two,Three,Four"']
};

```
it works, **but if the length of that single element in the array ('"One,Two,Three,Four"') is > 1024 (quotes included)** 

on LibreOffice:
it treats the entire string as a single value
on Microsoft excel:
it gives an error on opening the file and if you let it fix that it will remove all list validation

can we add more values without these issues?

**Comment by rohitvishwakarma1819** [comment]
@DevMoustafa97 did you find any solution ?

**Comment by Kiran-Oleti** [comment]
@DevMoustafa97 @rohitvishwakarma1819 did you find the solution to add n number of values to the dropdown without gettinng error in excel while opening

**Comment by kherkeladze** [comment]
You can fix this issue to move dropdown values to separate sheet and inject data from it. 
 
 
```
const helperSheet = workbook.addWorksheet('Helper');
data.forEach((item, index) => {
helperSheet.getCell('A${index + 1
    }').value = item;
});

// hide helper sheet  
helperSheet.state = 'hidden';

// Define named range
workbook.definedNames.add(
'Helper!$A$1:$A$' + data.ength,
'DataList',
);

worksheet.getRow(i).getCell('destinationCell').dataValidation = {
  type: 'list',
  allowBlank: true,
  formulae: ['=DataList'
    ],
  showErrorMessage: true,
  errorTitle: 'Invalid input',
  error: 'Please select from the list',
};
```
